backend/src/handlers/user_handler.go

```diff
@@ -324,3 +335,14 @@ func (srv *Server) HandleResetStudentPassword(w http.ResponseWriter, r *http.Req
 	}
 	srv.WriteResponse(w, http.StatusOK, response)
 }
+
+func validateUsername(username string) string {
```

Comment:
I was thinking we could tag the structs, and then be able to validate by calling `ValidateStruct()`
So something like
```go
// static variable in database/DB.go
var validate *validator.Validate


// models/users.go
type User struct {
DbFields
Username string `json:"username" validate:"alphanum"`
Email string `json:"email" validate:"email;omitempty"`
/...

err := validator.ValidateStruct(user)
```

I can make a ticket for this if you'd like

Reply:
I'm happy to do it, so if you think a ticket would be tidier than adding extra to this PR that'd be good.  Otherwise I can just add it to this PR.

Reply:
How about add validation to just the user in this PR in that manor, and I'll get a ticket made up for the rest?